### PR TITLE
Store: Check for plugin errors when setting up or fixing a store

### PR DESCRIPTION
**Pull request by ryelle**
As reported in #23899, the new "store fixer" step can get stuck in a perpetual loading state due to the `folder_exists` error (also reported in the setup flow, in #18462).

This appears to be a problem where a required plugin was removed from a site, but isn't completely gone. WordPress core is very careful to not overwrite an existing plugin, so it sees the remains of the other plugin and aborts the install. This causes an error, but also leaves the store in an inaccessible state, so now we're detecting any errors in installing, and directing the user to contact support.

We also try to let the user know which plugin had the problem, so they can pass that on to HEs.

screen shot 2018-04-06 at 5 08 05 pm

[See a screen recording](https://cloudup.com/cTZL0_XlXIt)

**To test**

First, break your site 😁 

- Do this on a site where you have FTP access
- Delete one of the [required plugins](https://github.com/Automattic/wp-calypso/blob/master/client/extensions/woocommerce/lib/get-required-plugins.js), for example woocommerce-services
- Add a folder to your plugins dir, `woocommerce-services`
- Add any file to that folder (WP is smart enough to install over an empty directory)

Now you can test this branch

- Go to any store page, `/store/:site`
- It should detect that you're missing a plugin, and try to install it
- It should fail, because you broke your site, and tell you to contact support
- Remove the fake plugin folder, and reload the store
- It should install the plugin correctly, and take you on to the requested page

Theoretically this also fixes this error in the setup flow, but I don't know how to replicate this for a site that hasn't been set up yet.

### CODE REVIEW

**Comment by timmyc** [comment]
Thanks for the quick fix on this one too @ryelle - very interesting edge case to track down, and not easy to reproduce on atomic sites.

File: client/extensions/woocommerce/app/dashboard/required-plugins-install-view.js [hunk 1]
```diff
@@ -395,6 +407,36 @@ class RequiredPluginsInstallView extends Component {
...
+		return (
+			<div className="dashboard__setup-wrapper setup__wrapper">
+				<div className="card dashboard__plugins-install-view">
+					<SetupHeader
+						imageSource={ '/calypso/images/extensions/woocommerce/woocommerce-store-creation.svg' }
+						imageWidth={ 160 }
+						title={ translate( "We can't update your store" ) }
+						subtitle={ translate(
+							"Your store is missing some required plugins. We can't fix this automatically due to " +
+								'a problem with the {{b}}%(plugin)s{{/b}} plugin. {{br/}}{{br/}}' +
```
Comment (akirk) [inline client/extensions/woocommerce/app/dashboard/required-plugins-install-view.js]: Could we rename `%(plugin)s` to `%(pluginName)` to make it more clear what the variable will contain?

Comment (akirk) [inline client/extensions/woocommerce/app/dashboard/required-plugins-install-view.js]: I would recommend to not include the double `<br/>` in the translatable string. Could you split the translate calls into two that only contain the translatable text? Thanks!

**Comment by ryelle** [comment]
@akirk thanks for the translation feedback 🙂

**Pull request merged**